### PR TITLE
Reintegrate skeleton-docker into repository

### PR DESCRIPTION
**Pull request by zenine07**
# 

## 🗣 Description

This PR would reintegrate the [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker) skeleton into this repository.

This was done by adding the correct skeleton repository as a remote, pulling from its develop branch, and resolving any conflicts shown. 

```console
git remote add skeleton https://github.com/cisagov/skeleton-docker.git
git remote set-url --push skeleton no_push
git pull --allow-unrelated-histories skeleton develop
git status
```

After skeleton was merged, fixed pre-commit issues that showed up.

## 💭 Motivation and Context

When this project was created, it was not started using the [`skeleton` tool](https://github.com/cisagov/development-guide/tree/develop/project_setup#skeleton-tool-) This prevents the [`lineage` GitHub Action](https://github.com/cisagov/action-lineage) from automatically generating PRs to ensure downstream repositories are kept up-to-date.

## 🧪 Testing

Automated tests pass.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by zenine07** [comment]
I've gotten everything to pass, except the "Build and push all platforms". I'm not sure whether this image needs to be pushed to Dockerhub, and if so, how I get the credentials in the repo to do it.

File: .github/CODEOWNERS
Comment (dav3r) [inline .github/CODEOWNERS]: Should we all be code owners until we figure out how we want to pare down this list?

```suggestion
* @dav3r @felddy @hillaryj @INLGuy @jsf9k @mcdonnnj @zenine07
```

File: .github/workflows/build.yml
Comment (dav3r) [inline .github/workflows/build.yml]:
```suggestion
      # docker-compose-check pre-commit hook to pass.
```

File: .github/workflows/build.yml
Comment (dav3r) [inline .github/workflows/build.yml]:
```suggestion
      # tests to run successfully.
```

**Review by dav3r — COMMENTED** [review]
> I've gotten everything to pass, except the "Build and push all platforms". I'm not sure whether this image needs to be pushed to Dockerhub, and if so, how I get the credentials in the repo to do it.

I don't think there's any reason why this image can't live in our public Dockerhub, right?  If that's the case, @felddy or @jsf9k can add the cisagov Dockerhub creds to this repo's secrets (`DOCKER_USERNAME` and `DOCKER_PASSWORD`).  Alternatively, if one of you provides them to me, I can do it.

File: CONTRIBUTING.md
Comment (dav3r) [inline CONTRIBUTING.md]:
```suggestion
issue](https://github.com/cisagov/con-pca-api/issues) in this
```

File: CONTRIBUTING.md
Comment (dav3r) [inline CONTRIBUTING.md]:
```suggestion
request](https://github.com/cisagov/con-pca-api/pulls), you will
```

File: CONTRIBUTING.md
Comment (dav3r) [inline CONTRIBUTING.md]:
```suggestion
cd con-pca-api
pyenv virtualenv <python_version_to_use> con-pca-api
pyenv local con-pca-api
```

File: requirements.txt
Comment (dav3r) [inline requirements.txt]: Wherever possible, you should document in this file why each package is pinned to a particular version.  If a package does not need to be pinned to a specific version, let it fly free!

File: src/api/utils/subscription/targets.py
Comment (dav3r) [inline src/api/utils/subscription/targets.py:63]: For improved context, please include the name/id of the bandit error in question here.

File: src/api/utils/template/templates.py
Comment (dav3r) [inline src/api/utils/template/templates.py:65]: For improved context, please include the name/id of the bandit error in question here.

**Review by dav3r — CHANGES_REQUESTED** [review]
I still have a few more files to review, but here are a few items to chew on in the meantime.

File: src/api/utils/subscription/targets.py
Comment (zenine07) [reply src/api/utils/subscription/targets.py:63]: I have provided a link in the comments for B311 - random. https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b311-random

File: src/api/utils/template/templates.py
Comment (zenine07) [reply src/api/utils/template/templates.py:65]: I have provided a link in the comments for B311 - random. https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b311-random

File: requirements.txt
Comment (zenine07) [reply requirements.txt]: I have added comments to requirements that cannot changed due to gophish version 0.5.1

**Comment by jsf9k** [comment]
> > I've gotten everything to pass, except the "Build and push all platforms". I'm not sure whether this image needs to be pushed to Dockerhub, and if so, how I get the credentials in the repo to do it.
> 
> I don't think there's any reason why this image can't live in our public Dockerhub, right? If that's the case, @felddy or @jsf9k can add the cisagov Dockerhub creds to this repo's secrets (`DOCKER_USERNAME` and `DOCKER_PASSWORD`). Alternatively, if one of you provides them to me, I can do it.

I don't know the Docker Hub creds we use for this.  I only know my personal creds.

File: requirements.txt
Comment (dav3r) [reply requirements.txt]: @zenine07 Are you able to remove the specific versions pinned to all of those other requirements (the ones not required by gophish)?  Or change them so that they apply to versions greater than or equal to the versions specified?

**Comment by felddy** [comment]
> > > I've gotten everything to pass, except the "Build and push all platforms". I'm not sure whether this image needs to be pushed to Dockerhub, and if so, how I get the credentials in the repo to do it.
> > 
> > 
> > I don't think there's any reason why this image can't live in our public Dockerhub, right? If that's the case, @felddy or @jsf9k can add the cisagov Dockerhub creds to this repo's secrets (`DOCKER_USERNAME` and `DOCKER_PASSWORD`). Alternatively, if one of you provides them to me, I can do it.
> 
> I don't know the Docker Hub creds we use for this. I only know my personal creds.

I've added the `DOCKER_USERNAME` and `DOCKER_PASSWORD` secrets to the organization and given this repository access to them.  (Organization-level secrets is a relatively new feature).  I believe this should now enable the workflow to push to the Docker hub.

File: requirements.txt
Comment (zenine07) [reply requirements.txt]: @dav3r I'd be willing to do ~= (minor updates) for all other requirements. I don't really trust allowing major updates.

Comment (dav3r) [reply requirements.txt]: I'll take it- that is an improvement over pinning to an exact version.  Our mode of operation has typically been to just roll with new versions and only pin when we find a problem.

Comment (zenine07) [reply requirements.txt]: I'll at least try >= for now, I believe the tests we have wrote will find any issues if there are any, I've just never liked trusting backwards-compatibility in major updates.

File: src/scripts/create_dummy_reporting_data.py
Comment (dav3r) [inline src/scripts/create_dummy_reporting_data.py:192]: You should include a similar comment here about the `# nosec` and can mention that it applies throughout this file.

File: src/scripts/create_dummy_reporting_data.py
Comment (dav3r) [inline src/scripts/create_dummy_reporting_data.py]: BTW, you have a typo in this variable name: `decpection_rate` --> `deception_rate` - make sure you change it throughout the file.

File: src/secrets/quote.txt
Comment (dav3r) [inline src/secrets/quote.txt]: This file and directory (inherited from the skeleton) can be deleted, unless you plan on using them for something.

File: tests/authentication/test_backend.py
Comment (dav3r) [inline tests/authentication/test_backend.py:34]: Please document the reason for the `# nosec` here, similar to other places.

File: tests/container_test.py
Comment (dav3r) [inline tests/container_test.py:44]: 👍 👍

File: src/scripts/create_dummy_reporting_data.py
Comment (jsf9k) [reply src/scripts/create_dummy_reporting_data.py:191]: Is `decpectation_rate` a thing, or is that a typo?

File: src/scripts/create_dummy_reporting_data.py
Comment (jsf9k) [reply src/scripts/create_dummy_reporting_data.py]: Ha!  I thought "decpectation" was a word I didn't know.

File: htmlcov/status.json
Comment (jsf9k) [inline htmlcov/status.json]: Is this file intended to be committed?  This looks like testing coverage output, although I may be wrong about that.

**Review by jsf9k — COMMENTED** [review]
I had one question, and I agree with all the changes @dav3r mentioned.

File: htmlcov/status.json
Comment (zenine07) [reply htmlcov/status.json]: Yeah, that's test coverage, I'll remove that.

Comment (zenine07) [reply htmlcov/status.json]: @jsf9k I removed the html coverage in another commit and merged it into this branch to reduce the amount of files changed to look at. So are html coverage is no longer being commited.

File: tests/container_test.py
Comment (zenine07) [reply tests/container_test.py:44]: @dav3r Thanks for looking at that. I was curious what your thoughts would be on it.

File: src/secrets/quote.txt
Comment (zenine07) [reply src/secrets/quote.txt]: @dav3r I have removed that directory as I have no plans for it.

File: src/scripts/create_dummy_reporting_data.py
Comment (zenine07) [reply src/scripts/create_dummy_reporting_data.py]: @dav3r Fixed that typo.

File: src/scripts/create_dummy_reporting_data.py
Comment (zenine07) [reply src/scripts/create_dummy_reporting_data.py:192]: @dav3r Added a comment at the beginning of that function detailing why all those nosecs exist. This script here is not used in production and is only for generating local data to test with.

File: tests/authentication/test_backend.py
Comment (zenine07) [reply tests/authentication/test_backend.py:34]: @dav3r, I put a comment at the top of the file detailing why nosec exists in multiple of the tests in this file.

File: requirements.txt
Comment (zenine07) [reply requirements.txt]: @dav3r I have added >= to all of the requirements. All of the automated tests were passing and also ran a few manual tests of my own and everything looks good using >=. I was able to get away with pinning specific versions as well.

File: src/scripts/create_dummy_reporting_data.py
Comment (dav3r) [reply src/scripts/create_dummy_reporting_data.py:192]: @zenine07 FYI, our convention is to have the person who initiated a GitHub conversation be the one to resolve it.  That gives the initiator the opportunity to ensure that they are satisfied with the resolution.

The main exception to this is if a code suggestion is made during a review; when you commit that suggestion, GitHub automatically resolves the conversation.  Otherwise, you shouldn't manually resolve a conversation that someone else started.

File: tests/container_test.py
Comment (dav3r) [reply tests/container_test.py:44]: If you can think of any other ways to validate that the containers have started up properly, feel free to add them.  Better yet, if something ever fails to start up properly in your testing, see if you can add a test that would detect that situation.

**Comment by dav3r** [comment]
@zenine07 - Have you tried to resolve the last remaining check that is failing?  I see this [error output](https://github.com/cisagov/con-pca-api/pull/215/checks?check_run_id=2221698953):
```
#75 1013.8   running build_rust
#75 1013.8   
#75 1013.8       =============================DEBUG ASSISTANCE=============================
#75 1013.8       If you are seeing a compilation error please try the following steps to
#75 1013.8       successfully install cryptography:
#75 1013.8       1) Upgrade to the latest pip and try again. This will fix errors for most
#75 1013.8          users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
#75 1013.8       2) Read https://cryptography.io/en/latest/installation.html for specific
#75 1013.8          instructions for your platform.
#75 1013.8       3) Check our frequently asked questions for more information:
#75 1013.8          https://cryptography.io/en/latest/faq.html
#75 1013.8       4) Ensure you have a recent Rust toolchain installed:
#75 1013.8          https://cryptography.io/en/latest/installation.html#rust
#75 1013.8       5) If you are experiencing issues with Rust for *this release only* you may
#75 1013.8          set the environment variable `CRYPTOGRAPHY_DONT_BUILD_RUST=1`.
#75 1013.8       =============================DEBUG ASSISTANCE=============================
#75 1013.8   
#75 1013.8   error: can't find Rust compiler
#75 1013.8   
#75 1013.8   If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.
#75 1013.8   
#75 1013.8   To update pip, run:
#75 1013.8   
#75 1013.8       pip install --upgrade pip
#75 1013.8   
#75 1013.8   and then retry package installation.
#75 1013.8   
#75 1013.8   If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs) is the recommended way to download and update the Rust compiler toolchain.
#75 1013.8   
#75 1013.8   This package requires Rust >=1.41.0.
#75 1013.8   ----------------------------------------
#75 1013.9   ERROR: Failed building wheel for cryptography
```

**Comment by zenine07** [comment]
@dav3r I was going to look at that today, not sure why that's failing all of a sudden.

**Comment by zenine07** [comment]
@dav3r I have moved that environment variable and added a reason as to why it's needed.

**Review by dav3r — APPROVED** [review]
This looks solid to me now- thanks @zenine07 for all of your hard work in getting this PR up to our rigorous standards!  🥇

File: requirements.txt
Comment (jsf9k) [inline requirements.txt:18]: This looks like testing-specific foo that should be in `requirements-test.txt`.

**Review by jsf9k — CHANGES_REQUESTED** [review]
Please address my one comment (I may be wrong), then re-request my review.  Also, there is a conflict that needs to be resolved before this PR can be merged..

File: requirements.txt
Comment (zenine07) [reply requirements.txt:18]: @jsf9k The ones that are highlighted are not testing specific. All of those libraries are used in the main application.

Comment (jsf9k) [reply requirements.txt:18]: Thanks!

**Comment by jsf9k** [comment]
@zenine07, please resolve the conflict and then I'm happy to approve.

**Review by jsf9k — APPROVED** [review]
The checks finally completed!  Approving with great gusto.  Nice work @zenine07!

**Comment by zenine07** [comment]
Thanks @jsf9k! @dav3r Do you have any idea why I can't merge the pull request now? Is it because build / Build and push all platforms (pull_request) was skipped?

**Comment by jsf9k** [comment]
> Thanks @jsf9k! @dav3r Do you have any idea why I can't merge the pull request now? Is it because build / Build and push all platforms (pull_request) was skipped?

@inlguy needs to review, since he was the only other original codeowner.  The new codeowners don't kick in until after this merge.

**Review by inlguy — APPROVED** [review]
Looks good.   There is some here from older commits.   I assume that is just a by product of the shuffle.  But looks good.

**Pull request merged**
